### PR TITLE
*: add parser and definition for all unsupported builtin funcs (part 6) 

### PR DESCRIPTION
**Pull request by XuHuaiyu**
This commit serves for the following builtin funcs:

ANY_VALUE, INET_ATON, INET_NTOA, INET6_ATON,
INET6_NTOA, IS_FREE_LOCK, IS_IPV4, IS_IPV4_COMPAT,
IS_IPV4_MAPPED, IS_IPV6, IS_USED_LOCK, MASTER_POS_WAIT,
NAME_CONST, RELEASE_ALL_LOCKS, UUID, UUID_SHORT

PTAL @shenli @coocood @hanfei1991 @zimulala @tiancaiamao

### CODE REVIEW

File: expression/builtin_miscellaneous.go [hunk 1]
```diff
@@ -0,0 +1,445 @@
...
+
+func (c *anyValueFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	return &builtinAnyValueSig{newBaseBuiltinFunc(args, ctx)}, errors.Trace(c.verifyArgs(args))
+}
+
+type builtinAnyValueSig struct {
+	baseBuiltinFunc
+}
+
+// https://dev.mysql.com/doc/refman/5.7/en/miscellaneous-functions.html#function_any-value
```
Comment (coocood) [inline expression/builtin_miscellaneous.go]: `See {URL}`

File: expression/builtin.go [hunk 2]
```diff
@@ -287,7 +287,24 @@ var funcs = map[string]functionClass{
...
+	ast.IsFreeLock:      &isFreeLockFunctionClass{baseFunctionClass{ast.IsFreeLock, 1, 1}},
+	ast.IsIPv4:          &isIPv4FunctionClass{baseFunctionClass{ast.IsIPv4, 1, 1}},
+	ast.IsIPv4Compat:    &isIPv4CompatFunctionClass{baseFunctionClass{ast.IsIPv4Compat, 1, 1}},
+	ast.IsIPv4Mapped:    &isIPv4MappedFunctionClass{baseFunctionClass{ast.IsIPv4Mapped, 1, 1}},
+	ast.IsIPv6:          &isIPv6FunctionClass{baseFunctionClass{ast.IsIPv6, 1, 1}},
+	ast.IsUsedLock:      &isUsedLockFunctionClass{baseFunctionClass{ast.IsUsedLock, 1, 1}},
+	ast.MasterPosWait:   &masterPosWaitFunctionClass{baseFunctionClass{ast.MasterPosWait, 2, 4}},
+	ast.NameConst:       &nameConstFunctionClass{baseFunctionClass{ast.NameConst, 2, 2}},
+	ast.ReleaseAllLocks: &releaseAllLocksFunctionClass{baseFunctionClass{ast.ReleaseAllLocks, 0, 0}},
+	ast.UUID:            &uuidFunctionClass{baseFunctionClass{ast.UUID, 1, 1}},
```
Comment (coocood) [inline expression/builtin.go]: UUID args should be 0.

File: expression/builtin_miscellaneous.go [hunk 3]
```diff
@@ -0,0 +1,445 @@
...
+import (
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/util/types"
+	"time"
+)
+
+var (
+	_ functionClass = &sleepFunctionClass{}
+	_ functionClass = &lockFunctionClass{}
```
Comment (tiancaiamao) [inline expression/builtin_miscellaneous.go]: Maybe we'll never implement this kind of function.
what's your opinion? @shenli

File: expression/builtin_miscellaneous_test.go [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Copyright 2017 PingCAP, Inc.
```
Comment (tiancaiamao) [inline expression/builtin_miscellaneous_test.go]: Is builtin function sleep moved from other place? Does it has test? if so, please also move test code.

File: expression/builtin_miscellaneous.go (hunk 3 above)
Comment (shenli) [inline expression/builtin_miscellaneous.go]: We could add those function as unimplemented first.

File: expression/builtin_miscellaneous_test.go (hunk 4 above)
Comment (XuHuaiyu) [inline expression/builtin_miscellaneous_test.go]: It does be moved from builtin_other.go, but there is no builtin_other_test.go.
We do test these functions in evaluator_test.go. @tiancaiamao

**Comment by XuHuaiyu** [comment]
@coocood @tiancaiamao PTAL

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**